contents/data_vis.md

```diff
@@ -368,9 +368,157 @@ What more can we ask for?
+See `?cgrad` for more information.
+
+```jl
+scolor = """
+    figure = (;resolution = (400,300), font= "CMU Serif")
+    axis = (; xlabel = L"x", ylabel = L"y", aspect= DataAspect())
+
+    fig, ax, pltobj = CairoMakie.heatmap(rand(20,20); colorrange = (0,1),
+        colormap = Reverse(:viridis), axis = axis, figure = figure)
+    Colorbar(fig[1,2], pltobj, label = "Reverse colormap Sequential")
+    fig
+    label = "Reverse_colormap_sequential" # hide
+    caption = "Reverse colormap sequential and colorrange." # hide
+    Options(fig; filename=label, label, caption) # hide
+    """
+sco(scolor)
+```
+
+When setting a `colorrange` usually the values outside this range are colored with the first and last color from  the colormap.
+However, sometimes is better to specified the color that you want at both ends. We do that with `highclip` and `lowclip`:
+
+```
+using ColorSchemes
+```
+
+```jl
+s = """
+    figure = (;resolution = (400,300), font= "CMU Serif")
+    axis = (; xlabel = L"x", ylabel = L"y", aspect= DataAspect())
+
+    fig, ax, pltobj = CairoMakie.heatmap(randn(20,20); colorrange = (-2,2),
+        colormap = "diverging_rainbow_bgymr_45_85_c67_n256",
+        highclip = :black, lowclip = :white, axis = axis, figure = figure)
+    Colorbar(fig[1,2], pltobj, label = "Diverging colormap")
+    fig
+    label = "diverging_colormap" # hide
+    caption = "Diverging Colormap with low and high clip." # hide
+    Options(fig; filename=label, label, caption) # hide
+    """
+sco(s)
+```
+
+But we mentioned that also RGB vectors are valid options.
+For our next example you could pass the custom colormap _perse_ or use `cgrad` to force a categorical colorbar.
+
+```
+using ColorSchemes
+```
+
+```jl
+scat = """
+    figure = (;resolution = (400,300), font= "CMU Serif")
+    axis = (; xlabel = L"x", ylabel = L"y", aspect= DataAspect())
+    cmap = ColorScheme(range(colorant"red", colorant"green", length=3))
+    mygrays = ColorScheme([RGB{Float64}(i, i, i) for i in [0.0,0.5,1.0]])
+
+    fig, ax, pltobj = CairoMakie.heatmap(rand(-1:1,20,20);
+        colormap = cgrad(mygrays, 3, categorical = true, rev = true), # cgrad and Symbol, mygrays,
+        axis = axis, figure = figure)
+    cbar = Colorbar(fig[1,2], pltobj, label = "Categories")
+    cbar.ticks = ([-0.66,0,0.66], ["-1","0","1"])
+    fig
+    label = "categorical_colormap" # hide
+    caption = "Categorical Colormap." # hide
+    Options(fig; filename=label, label, caption) # hide
+    """
+sco(scat)
+```
+
+Lastly, the ticks in the colorbar for the categorial case are not centered by default in each color.
+This is fixed by passing custom ticks, as in `cbar.ticks = (positions, ticks)`.
+The last situation is when passing a `tuple` of two colors to `colormap` as symbols, strings or a mix.
+You will get an interpolated colormap between these two colors.
+
+Also, `hex` coded colors are also accepted. So, on top or our heatmap let's put one semi-transparent point using this.
```

Comment:
Do we need a backtick in `hex`? Is is a type or function from Julia/Makie etc?

Reply:
it's a function from colours.